frontend/micro-ui/web/micro-ui-internals/packages/modules/hearings/src/pages/employee/InsideHearingMainPage.js

```diff
@@ -1,27 +1,82 @@
-import React from "react";
+import React, { useEffect, useRef, useState } from "react";
 import { useTranslation } from "react-i18next";
 import { useHistory, Link } from "react-router-dom";
+import { useQuery, useQueryClient } from "react-query";
 import { Header, ActionBar, SVG, SubmitBar, Card } from "@egovernments/digit-ui-react-components";
 import { Button, TextArea } from "@egovernments/digit-ui-components";
 import EvidenceHearingHeader from "./EvidenceHeader";
 import HearingSideCard from "./HearingSideCard";
+import debounce from 'lodash/debounce';
 
 const fieldStyle = { marginRight: 0 };
 
+const fetchLatestText = async () => {
+  return localStorage.getItem("latest-text");
+};
+
 const InsideHearingMainPage = () => {
   const history = useHistory();
+  const [immediateText, setImmediateText] = useState("this is good");
+  const [delayedText, setDelayedText] = useState("");
+  const [userRoles, setUserRoles] = useState([]);
+  const textAreaRef = useRef(null);
+  const queryClient = useQueryClient();
+
+  useEffect(() => {
+    const userDetails = JSON.parse(localStorage.getItem("user-info"));
+    setUserRoles(userDetails.roles);
+  }, []);
+
+  const checkUserApproval = (userRole) => {
+    return userRoles.some((role) => role.name === userRole);
+  };
+
+  const { data: latestText } = useQuery("latestText", fetchLatestText, {
+    refetchInterval: 5000,
+    enabled: !checkUserApproval("ADVOCATE_APPROVER"),
+  });
+
+  useEffect(() => {
+    if (latestText) {
+      setDelayedText(latestText);
+    }
+  }, [latestText]);
 
   const handleNavigate = (path) => {
     const contextPath = window?.contextPath || "";
     history.push(`/${contextPath}${path}`);
   };
 
+  const updateText = debounce((newText) => {
+    localStorage.setItem("latest-text", newText);
+    setDelayedText(newText);
+    queryClient.invalidateQueries("latestText");
```

Comment:
the person updating the text would see immediateText all the time, so query shouldn't run for them, AND, `queryClient.invalidateQueries` shouldn't need to be called.

Reply:
I see that the query is disabled, so I guess we can just remove the call to invalidateQueries